### ISSUE TITLE
Affiliation address format/labelling

### ISSUE THREAD

**Issue by Aazhar** [kermitt2/grobid#320]
I've remarked that the labelling for the affiliation address is different following the contain of the data feeded, for instance the presence of the country impacts whether a 'false positive' city name showing in the first part before the postcode here wilson is considered as settlement or not.

Output 1 :

    Laboratoire	laboratoire	L	La	Lab	Labo	e	re	ire	oire	LINESTART	INITCAP	NODIGIT	0	0	0	0	0	0	NOPUNCT	Xxxx	<affiliation>	I-<laboratory>
    d	d	d	d	d	d	d	d	d	d	LINEIN	NOCAPS	NODIGIT	1	0	0	0	0	0	NOPUNCT	x	<affiliation>	<laboratory>
    '	'	'	'	'	'	'	'	'	'	LINEIN	ALLCAPS	NODIGIT	1	0	0	0	0	0	QUOTE	'	<affiliation>	<laboratory>
    informatique	informatique	i	in	inf	info	e	ue	que	ique	LINEIN	NOCAPS	NODIGIT	0	0	0	0	0	0	NOPUNCT	xxxx	<affiliation>	<laboratory>
    Bâtiment	bâtiment	B	Bâ	Bât	Bâti	t	nt	ent	ment	LINEIN	INITCAP	NODIGIT	0	0	0	0	0	0	NOPUNCT	Xxxx	<affiliation>	<laboratory>
    d	d	d	d	d	d	d	d	d	d	LINEIN	NOCAPS	NODIGIT	1	0	0	0	0	0	NOPUNCT	x	<affiliation>	<laboratory>
    '	'	'	'	'	'	'	'	'	'	LINEIN	ALLCAPS	NODIGIT	1	0	0	0	0	0	QUOTE	'	<affiliation>	<laboratory>
    Alembert	alembert	A	Al	Ale	Alem	t	rt	ert	bert	LINEIN	INITCAP	NODIGIT	0	0	0	0	0	0	NOPUNCT	Xxxx	<affiliation>	I-<addrLine>
    61	61	6	61	61	61	1	61	61	61	LINEIN	NOCAPS	ALLDIGIT	0	0	0	0	0	0	NOPUNCT	dd	<affiliation>	<addrLine>
    Avenue	avenue	A	Av	Ave	Aven	e	ue	nue	enue	LINEIN	INITCAP	NODIGIT	0	0	1	0	0	0	NOPUNCT	Xxxx	<affiliation>	<addrLine>
    du	du	d	du	du	du	u	du	du	du	LINEIN	NOCAPS	NODIGIT	0	0	0	0	0	0	NOPUNCT	xx	<affiliation>	<addrLine>
    Président	président	P	Pr	Pré	Prés	t	nt	ent	dent	LINEIN	INITCAP	NODIGIT	0	0	0	0	0	0	NOPUNCT	Xxxx	<affiliation>	<addrLine>
    Wilson	wilson	W	Wi	Wil	Wils	n	on	son	lson	LINEIN	INITCAP	NODIGIT	0	1	0	0	1	0	NOPUNCT	Xxxx	<affiliation>	I-<settlement>
    94235	94235	9	94	942	9423	5	35	235	4235	LINEIN	NOCAPS	ALLDIGIT	0	0	0	0	0	0	NOPUNCT	dddd	<affiliation>	I-<postCode>
    cachan	cachan	c	ca	cac	cach	n	an	han	chan	LINEEND	NOCAPS	NODIGIT	0	0	0	0	1	0	NOPUNCT	xxxx	<affiliation>	I-<settlement>

Output 2 :

    Laboratoire	laboratoire	L	La	Lab	Labo	e	re	ire	oire	LINESTART	INITCAP	NODIGIT	0	0	0	0	0	0	NOPUNCT	Xxxx	<affiliation>	I-<laboratory>
    d	d	d	d	d	d	d	d	d	d	LINEIN	NOCAPS	NODIGIT	1	0	0	0	0	0	NOPUNCT	x	<affiliation>	<laboratory>
    '	'	'	'	'	'	'	'	'	'	LINEIN	ALLCAPS	NODIGIT	1	0	0	0	0	0	QUOTE	'	<affiliation>	<laboratory>
    informatique	informatique	i	in	inf	info	e	ue	que	ique	LINEIN	NOCAPS	NODIGIT	0	0	0	0	0	0	NOPUNCT	xxxx	<affiliation>	<laboratory>
    Bâtiment	bâtiment	B	Bâ	Bât	Bâti	t	nt	ent	ment	LINEIN	INITCAP	NODIGIT	0	0	0	0	0	0	NOPUNCT	Xxxx	<affiliation>	<laboratory>
    d	d	d	d	d	d	d	d	d	d	LINEIN	NOCAPS	NODIGIT	1	0	0	0	0	0	NOPUNCT	x	<affiliation>	<laboratory>
    '	'	'	'	'	'	'	'	'	'	LINEIN	ALLCAPS	NODIGIT	1	0	0	0	0	0	QUOTE	'	<affiliation>	<laboratory>
    Alembert	alembert	A	Al	Ale	Alem	t	rt	ert	bert	LINEIN	INITCAP	NODIGIT	0	0	0	0	0	0	NOPUNCT	Xxxx	<affiliation>	I-<addrLine>
    61	61	6	61	61	61	1	61	61	61	LINEIN	NOCAPS	ALLDIGIT	0	0	0	0	0	0	NOPUNCT	dd	<affiliation>	<addrLine>
    Avenue	avenue	A	Av	Ave	Aven	e	ue	nue	enue	LINEIN	INITCAP	NODIGIT	0	0	1	0	0	0	NOPUNCT	Xxxx	<affiliation>	<addrLine>
    du	du	d	du	du	du	u	du	du	du	LINEIN	NOCAPS	NODIGIT	0	0	0	0	0	0	NOPUNCT	xx	<affiliation>	<addrLine>
    Président	président	P	Pr	Pré	Prés	t	nt	ent	dent	LINEIN	INITCAP	NODIGIT	0	0	0	0	0	0	NOPUNCT	Xxxx	<affiliation>	<addrLine>
    Wilson	wilson	W	Wi	Wil	Wils	n	on	son	lson	LINEIN	INITCAP	NODIGIT	0	1	0	0	1	0	NOPUNCT	Xxxx	<affiliation>	<addrLine>
    94235	94235	9	94	942	9423	5	35	235	4235	LINEIN	NOCAPS	ALLDIGIT	0	0	0	0	0	0	NOPUNCT	dddd	<affiliation>	I-<postCode>
    cachan	cachan	c	ca	cac	cach	n	an	han	chan	LINEIN	NOCAPS	NODIGIT	0	0	0	0	1	0	NOPUNCT	xxxx	<affiliation>	I-<settlement>
    FR	fr	F	FR	FR	FR	R	FR	FR	FR	LINEEND	ALLCAPS	NODIGIT	0	0	0	0	0	0	NOPUNCT	XX	<affiliation>	<settlement>

Also, the county code are not labelled with country label.

**Comment by lfoppiano** [comment]
Some fixes related to this issue here: 13a6cfb9f720f946a3f7575940e76354179bf8e6 (to keep track)

**Comment by kermitt2** [comment]
This is not a surprising behaviour as such, we have a machine learning contextual model so two input where one is a substring of the other might have a different labelling, depending on the distribution of training data. I think we would need more example of countries names provides as 2-letter codes ("FR"), because usually we have the full names ("France") in the training data - we're in a typical overfitting case I think.
